engine/src/test/scala/cromwell/jobstore/JobStoreWriterSpec.scala

```diff
@@ -52,26 +57,26 @@ class JobStoreWriterSpec extends CromwellTestKitWordSpec with Matchers with Befo
       case JobStoreWriteSuccess(RegisterWorkflowCompleted(id)) => id shouldBe workflowId
       case message => fail(s"Unexpected response message: $message")
     }
-    jobStoreWriter.underlyingActor.stateName shouldBe Pending
     ()
   }
 
   "JobStoreWriter" should {
     "be able to collapse writes together if they arrive while a database access is ongoing" in {
 
-      sendRegisterCompletions(attempts = 3)
-
+      // Send a job completion. The database will hang.
+      sendRegisterCompletion(1)
+      // Wait long enough to know that the writer should be trying to write.
+      Thread.sleep(sleepMillis)
```

Comment:
I really don't like "just wait a long time, to make sure the concurrency will have happened" in unit tests.

Could you instead set the polling interval to effectively infinite, and instead send the internal control commands yourself?